ci/run_black.ps1

```diff
@@ -0,0 +1,23 @@
+#
+# Use PowerShell Core for any scripts so they are re-usable across windows/mac/linux.
+#
+
+$ExitCode = 0
+$GitDiff = git diff --name-only origin/master | Select-String -Pattern ".*\.py" | Select-String -NotMatch ".*/project_template/.*"
+# If there is no diff between master, then black everything.
+if ( $GitDiff -eq $null ) {
+    black --check .
+    if ($LastExitCode -ne 0) { $ExitCode = $LastExitCode }
+}
+# Else black just the diff.
+else {
+    black --check $GitDiff
+    if ($LastExitCode -ne 0) { $ExitCode = $LastExitCode }
+    # If the project_template changed, then black the testproject too.
+    $GitDiffTempl = Write-Output $GitDiff | Select-String -Pattern "^diff .*/project_template/.*"
```

Comment:
`$GitDiffTempl` will always be null in this scenario. The pattern of `Select-String` will never actually be found for two reasons: it won't match `^diff` because `$GitDiff` is using `--name-only` mode; and is filtering out `project_template` .

Please make a change to something in project_template locally and test that it actually tries to run black on `testproject/` in that case.